File: DSA_Template_Python/Algo_LCS.py
def longestCommonSubsequence(text1, text2):
    m, n = len(text1), len(text2)  # Lengths of the two texts
    dp = [[0] * (n + 1) for _ in range(m + 1)]  # Initialize DP table with all zeros

    # Fill the DP table
    for i in range(1, m + 1):
        for j in range(1, n + 1):
            if text1[i - 1] == text2[j - 1]:
                # Characters match, increment the diagonal value
                dp[i][j] = 1 + dp[i - 1][j - 1]
            else:
                # Characters do not match, take the max of left and top values
                dp[i][j] = max(dp[i - 1][j], dp[i][j - 1])

    return dp[m][n]  # LCS length is in the bottom-right cell

File: DSA_Template_CPP/Algo_LCS.cpp
#include <vector>
#include <string>
#include <algorithm>

using namespace std;

int longestCommonSubsequence(string text1, string text2) {
    int m = text1.size(); // Length of text1
    int n = text2.size(); // Length of text2
    vector<vector<int>> dp(m + 1, vector<int>(n + 1, 0)); // DP matrix

    // Building the DP matrix
    for (int i = 1; i <= m; i++) {
        for (int j = 1; j <= n; j++) {
            if (text1[i - 1] == text2[j - 1]) {
                // Characters match, increase the count
                dp[i][j] = 1 + dp[i - 1][j - 1];
            } else {
                // Characters do not match, take the max from left or top cell
                dp[i][j] = max(dp[i - 1][j], dp[i][j - 1]);
            }
        }
    }
    return dp[m][n]; // Length of LCS is in the bottom-right cell
}
